Sources/Perception/Internal/Memoization.swift

```diff
@@ -1,68 +1,43 @@
 import Foundation
 import OrderedCollections
 
-final class MemoizedCache<Key: Hashable, Value>: @unchecked Sendable {
-  private var dictionary = OrderedDictionary<Key, Value>()
-  private var lock = NSLock()
-  private let maxCapacity: Int
-  
-  // TODO possibly remove these stats later
-  private var totalEvictions = 0
-  private var totalCalls = 0
-  private var totalHits = 0
-  
+func memoize<Result>(
```

Comment:
Should we move all of this in a `#if DEBUG`?

Reply:
yep, done